src/osparc_control/core.py

```diff
@@ -0,0 +1,378 @@
+        )
+        self._continue: bool = True
+
+    def _sender_worker(self) -> None:
+        self._sender_receiver_pair.sender_init()
+
+        while self._continue:
+            message: Optional[
+                Union[CommandRequest, CommandReceived, CommandReply]
+            ] = self._out_queue.get()
+            if message is None:
+                # exit worker
+                break
+
+            # send message
+            self._sender_receiver_pair.send_bytes(message.to_bytes())
+
+        self._sender_receiver_pair.sender_cleanup()
+
+    def _handle_command_request(self, response: bytes) -> None:
+        command_request: Optional[CommandRequest] = CommandRequest.from_bytes(response)
+        if command_request is None:
+            return  # pragma: no cover
+
+        def _refuse_and_return(error_message: str) -> None:
+            self._out_queue.put(
+                CommandReceived(
+                    request_id=command_request.request_id,  # type: ignore
+                    accepted=False,
+                    error_message=error_message,
+                )
+            )
+            return
+
+        # check if command exists
+        if command_request.action not in self._exposed_interface:
+            error_message = (
+                f"No registered command found for action={command_request.action}. "
+                f"Supported actions {list(self._exposed_interface.keys())}"
+            )
+            _refuse_and_return(error_message)
+
+        manifest = self._exposed_interface[command_request.action]
+
+        # check command_type matches the one declared in the manifest
+        if command_request.command_type != manifest.command_type:
+            error_message = (
+                f"Incoming request command_type {command_request.command_type} "
+                f"do not match manifest's command_type {manifest.command_type} "
+                f"for command {command_request.action}"
+            )
+            _refuse_and_return(error_message)
+
+        # check if provided parametes match manifest
+        incoming_params_set = set(command_request.params.keys())
+        manifest_params_set = set(manifest._remapped_params.keys())
+        if incoming_params_set != manifest_params_set:
+            error_message = (
+                f"Incoming request params {command_request.params} do not match "
+                f"manifest's params {manifest.params}"
+            )
+            _refuse_and_return(error_message)
+
+        # accept command
+        self._out_queue.put(
+            CommandReceived(
+                request_id=command_request.request_id,
+                accepted=True,
+                error_message=None,
+            )
+        )
+
+        self._incoming_request_tracker.append(command_request)
+
+    def _handle_command_reply(self, response: bytes) -> None:
+        command_reply: Optional[CommandReply] = CommandReply.from_bytes(response)
+        assert command_reply  # noqa: S101
```

Comment:
what means noqa: s101???

Reply:
It tells `flake8` to ignore the error `S101` related to the assert (the description should be something similar to "Use of assert detected")